client/lib/posts/post-list-store.js

```diff
@@ -157,14 +159,20 @@ export default function( id ) {
 			_activeList.isLastPage = true;
 		}
 
-		posts = data.posts;
+		posts = sort( data.posts );
 
 		postIds = posts.map( function( post ) {
 			return post.global_ID;
 		} );
 
+		// were some posts missing from the response?
+		const removedPosts = getRemovedPosts( _activeList.postIds, postIds );
```

Comment:
I think this needs to be moved down to the `if( postIds.length )` block to handle an empty array